pkg/pool-weighted/test/ManagedPool.test.ts

```diff
@@ -648,10 +670,24 @@ describe('ManagedPool', function () {
           await pool.addAllowedAddress(owner, other.address);
         });
 
-        it('the listed LP can join', async () => {
-          await pool.joinAllGivenOut({ from: other, bptOut: FP_ONE });
+        context('when joins are enabled', () => {
+          it('the listed LP can join', async () => {
+            await pool.joinAllGivenOut({ from: other, bptOut: FP_ONE });
+
+            expect(await pool.balanceOf(other)).to.be.gt(0);
+          });
+        });
+
+        context('when joins are disabled', () => {
+          sharedBeforeEach(async () => {
+            await pool.setJoinExitEnabled(owner, false);
+          });
 
-          expect(await pool.balanceOf(other)).to.be.gt(0);
+          it('reverts', async () => {
+            await expect(pool.joinAllGivenOut({ from: other, bptOut: FP_ONE })).to.be.revertedWith(
+              'JOINS_EXITS_DISABLED'
+            );
```

Comment:
The tests are unfortunally not too clear. What was being tested here was that the allowlist works (it should have e.g. `describe('allowlist')` - which can be gleaned from the fact that only `joinAllGivenOut` is tested.

A better place to put this test is under `describe('proportional joins')` and `describe('disporportionate joins')`, so that we test this feature works for _all_ join (and exit) kinds.

Reply:
Sounds good.
The current test makes the split with swaps disabled, placing proportionate and disproportionate joins / exits inside.

Following that line I think it makes sense to add a new context and repeat the join / exit kinds: https://github.com/balancer-labs/balancer-v2-monorepo/pull/2070/commits/99b7f338157629a5586948f542dd1e4702ba282c